hr_expense_cancel/models/hr_expense.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    def action_cancel(self):
+        for sheet in self:
+            account_moves = sheet.account_move_ids
+            sheet.account_move_ids = False
+            payments = sheet.payment_ids.filtered(lambda line: line.state != "cancel")
+            # case : cancel invoice from hr_expense
+            self._remove_reconcile_hr_invoice(account_moves)
+            # If the sheet is paid then remove payments
+            if sheet.state in ("done", "approve"):
+                if sheet.expense_line_ids[:1].payment_mode == "own_account":
+                    self._remove_move_reconcile(payments, account_moves)
+                payments.action_draft_cancel()
+            # Deleting the Journal entry if in the previous steps
+            # (if the expense sheet is paid and payment_mode == 'own_account')
+            # it has not been deleted
+            if account_moves.exists():
+                moves_to_cancel = account_moves.filtered(
+                    lambda move: move.state != "draft"
+                )
+                if moves_to_cancel:
+                    moves_to_cancel.button_cancel()
+                account_moves.with_context(force_delete=True).unlink()
```

Comment:
```suggestion
                account_moves.with_context(force_delete=True).unlink()
```

In my opinion, some accountants may not be comfortable with deleting documents. Instead, the move should be reversed for audit purposes.
Should this line include a new method as a hook?

Reply:
Maybe we should include this improve after the migration. What do you think?